### ISSUE TITLE
Unexpected image size differences

### ISSUE THREAD

**Issue by kylemh** [palmerhq/cypress-image-snapshot#82]
Hi! I posted an issue earlier this morning https://github.com/palmerhq/cypress-image-snapshot/issues/79 which you resolved by having a more helpful error message.

```
Error: Image size (1280x1162) different than saved snapshot size (1280x1161).
```

I saw #67, but I'm unsure if it's related, as I'm using `cypress run` both locally and in CI. Feel to close, and I'll copy-paste my input there.

I have no idea how I'm getting a 1 pixel difference on a static page, especially since the other static pages DONT have this difference. Do you have any ideas?

**Comment by crosscompile** [comment]
I think this is a different issue than #67 as it's not related to image size differences when using `cypress run` and `cypress open`.

It's tough to say for sure what the issue could be, we've seen some subtle OS differences creep in before (https://github.com/palmerhq/cypress-image-snapshot/issues/71).

A possible workaround could be to explicitly set the viewport with `cy.viewport` before calling `cy.matchImageSnapshot`:

```js
cy.viewport(1280, 1162);
cy.matchImageSnapshot();
```

**Comment by kylemh** [comment]
My understanding is that `cypress run` doesn't respect viewport calls after reading [this issue](https://github.com/cypress-io/cypress/issues/3324)

**Comment by crosscompile** [comment]
I think `cypress run` respects `cy.viewport()` calls, we use this approach to save different mobile and desktop screenshots.  It looks like the issue in https://github.com/cypress-io/cypress/issues/3324 is that Cypress doubles the resolution with `cypress open` which we have definitely had problems with.

It does seem like you've found an issue with `cy.screenshot()` that it differs in size on CI vs. local, I would open an issue with Cypress regarding it.

In the meantime I would at least try explicitly calling `cy.viewport()`, or if you want to try running your Cypress tests in a container that can guarantee more consistency, they have [some docs on containers](https://docs.cypress.io/examples/examples/docker.html#Images).

**Comment by kylemh** [comment]
My tests are running in Travis which is essentially a container, no?

I tried explicitly calling viewport() in my custom command to no avail. I’ll go and write an issue on their repository. Strange that it’s only these two specific pages.

**Comment by crosscompile** [comment]
That is strange behavior, it definitely seems like a bug if you're setting a specific viewport size and the screenshot is a different size.

It looks like Travis has a few [virtual machine options](https://docs.travis-ci.com/user/reference/overview/), but the goal with running Cypress in a container would be to use the same operating system locally and in CI.  For example if you were using a Travis Ubuntu environment, unless your local machine is also Ubuntu there could be subtle differences in the environments.

**Comment by kylemh** [comment]
You've given me a lot to go on, and I don't think my issue is related to `cypress-image-snapshot`, so I'm gonna close it. I'll investigate more and see if containerization eliminates my issue. Will repot to cypress' issues if I can't resolve the issue.

**Comment by kylemh** [comment]
Looks like `cy.viewport()` doesn't behave predictably in headless: https://github.com/cypress-io/cypress/issues/2102#issuecomment-402901382

**Comment by edelgado** [comment]
I'm having this issue as well:

```
Error: Image size (1872x1256) different than saved snapshot size (936x628).
```

Screenshots taken in `cypress run` mode are 2x as big as the ones taken via `cypress open` mode.

**Comment by kylemh** [comment]
That seems potentially relevant to the resolution issue that's discussed on the Cypress issues. 

https://github.com/cypress-io/cypress/issues/2102

**Comment by danvoyce** [comment]
I'm also consistently able to replicate a similar issue, but it fails when I have the dev tools open in Chrome...

I take my snapshot with a fixed viewport size:

```
cy.viewport(1200, 800);
cy.get('[data-cy=canvas-preview]').matchImageSnapshot();
```

With the dev tools closed it's fine. But when I run the test with them open it fails

Seems the dev tools is eating up some of the space...

(docked to the right)
`Error: Image size (1150x1009) different than saved snapshot size (1070x1009).`

(docked to the bottom)
`Error: Image size (1150x1009) different than saved snapshot size (1150x655).`

**Comment by neirongkuifa** [comment]
To anyone who may encounter this issue: https://github.com/cypress-io/cypress/issues/3324#issuecomment-600125036
TLDR: Try `cypress run` instead of `cypress open` locally to generate base snapshots.

**Comment by adityabad** [comment]
Is this issue fixed? any updates? also i feel in my case it was caching the output image of a different run. when i re installed the plugin in a different folder it worked well

**Comment by Deop** [comment]
Having the same issue, with 1 pixel height difference. Tried mentioned workarounds with viewport and generating base image with `cypress run` but nothing helped.

**Comment by alex-vungle** [comment]
I am still experiencing this issue when use `cypress run` .

Cypress version: v8.0.0
Cypress Image Snapshot: v4.0.1

**Comment by rgangal** [comment]
I am also experiencing the same issue. It seems like `matchImageSnapshot` is not using the viewport to set the size of the image.

In my code I am setting the viewport and taking the screesshot: 
```
cy.viewport(1280, 1162);
cy.matchImageSnapshot()
```

but its saving the image with `1889x2060` when using `cypress run` on my local machine and saving the image with `944x1030` when running on jenkins.

**Comment by kylemh** [comment]
To everybody commenting in here. The issue is unrelated to this library. Please see https://docs.cypress.io/api/plugins/browser-launch-api#Set-screen-size-when-running-headless for a workaround and https://github.com/cypress-io/cypress/issues/587#issuecomment-812115782 for context.

I'm going to unsubscribe from this.

**Comment by LakshmiSanikommu** [comment]
> To anyone who may encounter this issue: [cypress-io/cypress#3324 (comment)](https://github.com/cypress-io/cypress/issues/3324#issuecomment-600125036) TLDR: Try `cypress run` instead of `cypress open` locally to generate base snapshots.

is this issue fixed now , or do we need to follow and run only **cypress open** which run in CLI ?
